backend/grant/user/views.py

```diff
@@ -344,10 +343,10 @@ def get_user_settings(user_id):
 
 @blueprint.route("/<user_id>/settings", methods=["PUT"])
 @auth.requires_same_user_auth
-# TODO guard all (shape, validity)
 @body({
     "emailSubscriptions": fields.Dict(required=False, missing=None),
-    "refundAddress": fields.Str(required=False, missing=None)
+    "refundAddress": fields.Str(required=False, missing=None,
+                                validate=lambda r: blockchain_get('/validate/address', {'address': r}))
```

Comment:
This doesn't actually work, it OKs it even if the response is `{"valid": False}`. You should probably make a dedicated  validator function that catches any errors in communication, and checks if `res.valid` to make sure the error is human-readable.

Reply:
Ooof, thanks for checking this.